test/integration/src/main/java/io/pravega/test/integration/selftest/adapters/ClientReader.java

```diff
@@ -151,11 +153,12 @@ private StreamReader getReader(String streamName) {
         StreamReader(String streamName) {
             this.readerGroup = UUID.randomUUID().toString().replace("-", "");
             this.readerId = UUID.randomUUID().toString().replace("-", "");
-            try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(ClientAdapterBase.SCOPE,
-                    ClientConfig.builder().controllerURI(ClientReader.this.controllerUri)
-                            .trustStore("../../config/cert.pem")
-                            .credentials(new DefaultCredentials("1111_aaaa", "admin"))
-                            .validateHostName(false).build())) {
+            ClientConfig clientConfig = ClientConfig.builder().controllerURI(ClientReader.this.controllerUri)
+                    .trustStore("../../config/cert.pem")
+                    .credentials(new DefaultCredentials("1111_aaaa", "admin"))
+                    .validateHostName(false).build();
+            ConnectionFactory connectionFactory = new ConnectionFactoryImpl(clientConfig);
```

Comment:
`ClientFactory` implements `AutoCloseable`. You need to keep a reference to this object and close it when appropriate

Reply:
Please fix this everywhere in your pull request. 

I have also seen instances of `ReaderGroupManager` not being closed when done with.